src/test/java/seedu/address/model/person/PersonTest.java

```diff
@@ -31,23 +32,78 @@ public void isSamePerson() {
+
+        // different matriculation number, all the other attributes same --> returns true
+        editedAlice = new PersonBuilder(ALICE).withMatricNumber(VALID_MATRICULATION_NUMBER_BOB).build();
+        assertTrue(ALICE.isSamePerson(editedAlice));
+
+        // different email, all the other attributes the same --> returns true
+        editedAlice = new PersonBuilder(ALICE).withEmail(VALID_EMAIL_BOB).build();
+        assertTrue(ALICE.isSamePerson(editedAlice));
+
+        // different phone number and matriculation, all other attributes same --> returns true
+        editedAlice = new PersonBuilder(ALICE).withPhone(VALID_PHONE_BOB)
+                .withMatricNumber(VALID_MATRICULATION_NUMBER_BOB).build();
+        assertTrue(ALICE.isSamePerson(editedAlice));
+
+        // different phone number and email, all other attributes same --> returns true
+        editedAlice = new PersonBuilder(ALICE).withPhone(VALID_PHONE_BOB).withEmail(VALID_EMAIL_BOB).build();
         assertTrue(ALICE.isSamePerson(editedAlice));
 
-        // different name, all other attributes same -> returns false
-        editedAlice = new PersonBuilder(ALICE).withName(VALID_NAME_BOB).build();
+        // different matriculation number and email, all other attributes same --> returns true
+        editedAlice = new PersonBuilder(ALICE).withMatricNumber(VALID_MATRICULATION_NUMBER_BOB)
+                .withEmail(VALID_EMAIL_BOB).build();
+        assertTrue(ALICE.isSamePerson(editedAlice));
+
+        // different phone, matriculation number and email, all the other attributes the same --> returns false
+        editedAlice = new PersonBuilder(ALICE).withPhone(VALID_PHONE_BOB)
+                .withMatricNumber(VALID_MATRICULATION_NUMBER_BOB).withEmail(VALID_EMAIL_BOB).build();
         assertFalse(ALICE.isSamePerson(editedAlice));
 
-        // name differs in case, all other attributes same -> returns false
-        Person editedBob = new PersonBuilder(BOB).withName(VALID_NAME_BOB.toLowerCase()).build();
-        assertFalse(BOB.isSamePerson(editedBob));
+        // email differs in case, all other attributes same -> returns true
+        Person editedBob = new PersonBuilder(BOB).withEmail(VALID_EMAIL_BOB.toUpperCase()).build();
+        assertTrue(BOB.isSamePerson(editedBob));
+
+        // matriculation number differs in case, all other attributes same -> returns true
+        editedBob = new PersonBuilder(BOB).withMatricNumber(VALID_MATRICULATION_NUMBER_BOB.toUpperCase()).build();
+        assertTrue(BOB.isSamePerson(editedBob));
+
+        editedBob = new PersonBuilder(BOB).withMatricNumber(VALID_MATRICULATION_NUMBER_BOB.toLowerCase()).build();
+        assertTrue(BOB.isSamePerson(editedBob));
+
+        // First character is lower case last character is upper case eg. a0253625M
+        String editedMatriculationNumber = VALID_MATRICULATION_NUMBER_BOB.substring(0, 1).toUpperCase()
+                + VALID_MATRICULATION_NUMBER_BOB.substring(1).toLowerCase();
```

Comment:
The comment for this should be the first character to uppercase and the rest to lowercase 👍

Reply:
Thanks for pointing out 😄